src/xtgeo/grid3d/_egrid.py

```diff
@@ -691,19 +704,21 @@ def map_axis_units(self) -> Units:
         return self.egrid_head.mapunits
 
     @map_axis_units.setter
-    def map_axis_units(self, value):
+    def map_axis_units(self, value: Units) -> None:
         self.egrid_head.mapunits = value
 
     @property
     def grid_units(self) -> Units:
+        assert self.egrid_head.gridunit is not None
         return self.egrid_head.gridunit.unit
 
     @grid_units.setter
-    def grid_units(self, value):
+    def grid_units(self, value: Units) -> None:
+        assert self.egrid_head.gridunit is not None
         self.egrid_head.gridunit.unit = value
 
     @classmethod
-    def from_xtgeo_grid(cls, xtgeo_grid):
+    def from_xtgeo_grid(cls, xtgeo_grid: GridProperty) -> EclGrid:
```

Comment:
I'm not 100% sure on this one, looking at the attributes used in the function it has to be GridProperty. @jcrivenaes Can you verify?

Reply:
`GridLike: Grid | GridProperty` is a thing too, maybe that applies?

Reply:
> `GridProperty`

Dont think so, this class does not have the _xtgformat2() functions, from what i can tell all the attributes called in the `from_xtgeo_grid(...)` exists on the Grid class. So i think @jcrivenaes statement is correct.

Reply:
> `GridLike: Grid | GridProperty` is a thing too, maybe that applies?

Dont think so, this class does not have the _xtgformat2() functions, from what i can tell all the attributes called in the from_xtgeo_grid(...) exists on the Grid class. So i think @jcrivenaes statement is correct.